### PR TITLE
feat: [PIE-4207]: Added ServiceStepBasic Info and DeployServiceDefinition Test case

### PR DESCRIPTION
**Pull request by vikrant-harness**
##### Summary:

Add ServiceStepBasic Info and DeployServiceDefinition Test case

##### Jira Links:

https://harness.atlassian.net/browse/PIE-4207

##### Screenshots:

Screenshot 2022-06-21 at 1 38 15 PM
Screenshot 2022-06-21 at 1 36 24 PM

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by vikrant-harness** [comment]
retrigger titlecheck

File: src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/__test__/DeployServiceDefinition.test.tsx [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+        }}
+      >
+        <ServiceContext.Provider value={serviceContextData as ServiceContextValues}>
+          <PipelineContext.Provider value={pipelineMockDataWithNativeHelm}>
+            <DeployServiceDefinition />
+          </PipelineContext.Provider>
+        </ServiceContext.Provider>
+      </TestWrapper>
+    )
+    // expect(container).toMatchSnapshot()
```
Comment (bhavya-sinha) [inline src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/__test__/DeployServiceDefinition.test.tsx:107]: commented code?

File: src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/__test__/DeployServiceDefinition.test.tsx [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    }),
+    updatePipeline: jest.fn(() => undefined)
+  } as any
+}
+
+const pipelineMockData = getContextValue(mockStageReturnWithoutManifestData)
+
+describe('DeployServiceDefinition', () => {
+  test('should render DeployServiceDefinition and switch between deployment type without opening dialog (no manifest/artifact data)', async () => {
+    const { container } = render(
```
Comment (udayvunnam) [inline src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/__test__/DeployServiceDefinition.test.tsx:41]: we can extract to common `renderDeployServiceDefinition` function

**Review by bhavya-sinha — APPROVED** [review]

File: src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/__test__/DeployServiceDefinition.test.tsx (hunk 1 above)
Comment (vikrant-harness) [reply src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/__test__/DeployServiceDefinition.test.tsx]: @bhavya-sinha  removed the same

File: src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/__test__/DeployServiceDefinition.test.tsx [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+        </ServiceContext.Provider>
+      </TestWrapper>
+    )
+    expect(container).toMatchSnapshot()
+    await waitFor(() => expect(container.querySelector('[data-icon="service-helm"]')).toBeInTheDocument())
+    const nativeHelmContainer = container.querySelector('[data-icon="service-helm"]')?.parentElement?.parentElement
+    fireEvent.click(nativeHelmContainer!)
+    const form = findDialogContainer() as HTMLElement
+    await waitFor(() => expect(form).toBeTruthy())
+    const confirmButton = (await findByText(form, 'confirm')) as HTMLElement
```
[the remark this answers is not in the document]
Comment (vikrant-harness) [reply src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/__test__/DeployServiceDefinition.test.tsx]: @udayvunnam  addressed the same in the latest commit

File: src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/__test__/DeployServiceDefinition.test.tsx (hunk 2 above)
Comment (vikrant-harness) [reply src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/__test__/DeployServiceDefinition.test.tsx]: @udayvunnam extracted in the latest commit

**Comment by udayvunnam** [comment]
LGTM. FYI good resource for testing with RTL https://kentcdodds.com/blog/common-mistakes-with-react-testing-library

**Review by udayvunnam — APPROVED** [review]

**Review by bhavya-sinha — APPROVED** [review]

**Pull request merged**